src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -1950,6 +1950,17 @@ private async Task CreateContainerAsync(RenderedModelResource cr, ILogger resour
 
             (spec.RunArgs, var failedToApplyRunArgs) = await BuildRunArgsAsync(resourceLogger, modelContainerResource, cancellationToken).ConfigureAwait(false);
 
+            // If a platform is specified in the container image annotation, add --platform to the run args.
+            if (modelContainerResource.TryGetContainerImageName(out var imageName) &&
```

Comment:
You shouldn't need this call to `TryGetContainerImageName`; it should be enough to simply retrieve the `ContainerImageAnnotation`.